### ISSUE TITLE
opkg distribute package failing

### ISSUE THREAD

**Issue by ldnelso2** [arduino/openwrt-packages-yun#32]
Hi Martino - 

You seem to have quite a bit of experience with managing packages for the yun.  I have been stuck for days on an issue with the package 'distribute.'  People on the arduino forums are having the same problem as well. 

I posted my question on the [AWS SDK repo](https://github.com/aws/aws-iot-device-sdk-arduino-yun/issues/73), but really it isn't an issue with the AWS SDK, it is a architectural conflict for the 'distribute' package on opkg install.  

In a nutshell:

$ opkg install distribute
Unknown package 'distribute'.
Collected errors:
 * opkg_install_cmd: Cannot install package distribute.

(I have already done an opkg update)

If you were having a Would your advice be to re-install a new fresh image of the yun rev2 firmware, or to figure out how to somehow cross compile the package for the mips_24k architecture?

**Comment by facchinm** [comment]
Hi @ldnelso2 ,
`distribute` in fact is not compiled nor present in the repos. There's, however, a recipe for aws sdk already packaged (you can grab it with `opkg install aws-iot-runtime-yun` and `opkg install aws-iot-device-sdk`). It's not bleeding edge but could suffice.
In case you need the latest and best, [this](https://github.com/arduino/openwrt-arduino-packages) is the repo you need to clone to start modifying the makefiles (you'll need the cross toolchain from http://downloads.arduino.cc/openwrtyun/17.11/packages/targets/ar71xx/generic/ledeyun-sdk-17.11-ar71xx-generic_gcc-7.3.0_musl.Linux-x86_64.tar.xz and these instruction https://openwrt.org/docs/guide-developer/using_the_sdk .
Hope it helps! If you have any further question feel free to write here :wink:

**Comment by ldnelso2** [comment]
Thank you, that does help.  For now, I am just going to forget about installing and doing anything with python on the yun, since not having easy setup or pip somewhat defeats the point.  

In terms of the repos, the package distribute is listed in the yun repo hosted by arduino.  

http://downloads.arduino.cc/openwrtyun/1/packages/distribute_0.6.21-1_ar71xx.ipk

That is what threw me off for so long.  Looks like somehow someone uploaded that incorrectly.  I will post that on their boards, b/c it looks to be confusing plenty of people. 

Thanks for the help!

**Comment by facchinm** [comment]
@ldnelso2 the `distribute` package you are linking has been built for an ancient version of openwrt so it won't surely work (notice the `/1/` in the url).
My advise here is: never install random ipk from the internet, always trust `opkg`.
Secondly, `pip` is available in the repos, just an `opkg update && opkg install python-pip` away, if there's enough space on the yun flash (otherwise you'll need to expand the filesystem using an sd card and [this](https://github.com/arduino/yun-go-updater/tree/master/YunDiskSpaceExpander) sketch).

**Comment by WhidbeyBill** [comment]
https://openwrt.org/toh/hwdata/arduino.cc/arduino.cc_yun
WELL - Got adventuresome and bricked my Rev2 Yun, no boot, gibberish from the serial terminal. SO **NO Not working for me.** **openwrt-19.07.1-ar71xx-generic-arduino-yun-squashfs-sysupgrade.bin**
